cfgov/v1/models/base.py

```diff
@@ -186,7 +186,16 @@ def specific_categories_query(block, parent_slug):
                         search_query
                     ).distinct().exclude(id=self.id).order_by(
                         '-date_published'
-                    )[:block.value['limit']])
+                    ))
+                # Apply similar logic as snippets.py's filter_by_tags method
+                # to enable AND filtering
+                if block.value['and_filtering']:
+                    for tag in self.tags.names() or []:
+                        related[search_type_name] = (
+                            related[search_type_name].filter(tags__name=tag)
+                        )
+                related[search_type_name] = \
+                    related[search_type_name][:block.value['limit']]
```

Comment:
Whoops, didn't thread my first reply to this. Anyway, the reason for this is probably because IntegerBlock wasn't added until Wagtail 1.6.